cns/middlewares/k8sSwiftV2.go

```diff
@@ -135,150 +133,78 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
-		if err != nil {
-			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
-		}
-		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
-		}
-
-		// Get and parse podCIDRs from env
-		podCIDRs, err := configuration.PodCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get podCIDRs from env")
-		}
-		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podCIDRs")
-		}
+	var podIPInfos []cns.PodIpInfo
 
-		// Get and parse serviceCIDRs from env
-		serviceCIDRs, err := configuration.ServiceCIDRs()
+	if len(mtpnc.Status.InterfaceInfos) == 0 {
+		// Use fields from mtpnc.Status if InterfaceInfos is empty
+		ip, prefixSize, err := utils.ParseIPAndPrefix(mtpnc.Status.PrimaryIP)
 		if err != nil {
-			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
-		}
-		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse serviceCIDRs")
-		}
-		// Check if the podIPInfo is IPv4 or IPv6
-		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
-		}
-		if ip.Is4() {
-			// routes for IPv4 podCIDR traffic
-			for _, podCIDRv4 := range podCIDRsV4 {
-				podCIDRv4Route := cns.Route{
-					IPAddress:        podCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv4Route)
-			}
-			// route for IPv4 serviceCIDR traffic
-			for _, serviceCIDRv4 := range serviceCIDRsV4 {
-				serviceCIDRv4Route := cns.Route{
-					IPAddress:        serviceCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv4Route)
-			}
-			// route for IPv4 infraVNETCIDR traffic
-			for _, infraVNETCIDRv4 := range infraVNETCIDRsv4 {
-				infraVNETCIDRv4Route := cns.Route{
-					IPAddress:        infraVNETCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, infraVNETCIDRv4Route)
+			return nil, errors.Wrap(err, "failed to parse mtpnc primary IP and prefix")
+		}
+		if prefixSize != prefixLength {
+			return nil, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
+		}
+
+		podIPInfos = append(podIPInfos, cns.PodIpInfo{
+			PodIPConfig: cns.IPSubnet{
+				IPAddress:    ip,
+				PrefixLength: uint8(prefixSize),
+			},
+			MacAddress:        mtpnc.Status.MacAddress,
+			NICType:           cns.DelegatedVMNIC,
+			SkipDefaultRoutes: false,
+			// InterfaceName is empty for DelegatedVMNIC
+		})
+	} else {
+		// Use InterfaceInfos if not empty
+		podIPInfos = make([]cns.PodIpInfo, len(mtpnc.Status.InterfaceInfos))
+		for i, interfaceInfo := range mtpnc.Status.InterfaceInfos {
```

Comment:
add tests for cases where mtpnc have multiple interfaceInfos? Right now I'm only seeing tests for mtpnc with single interfaceInfos